server/service/src/sync/test/test_data/goods_received.rs

```diff
@@ -30,8 +28,28 @@ const GOODS_RECEIVED: (&str, &str) = (
     }"#,
 );
 
-pub(crate) fn test_pull_upsert_records() -> Vec<TestSyncIncomingRecord> {
-    vec![TestSyncIncomingRecord::new_pull_upsert(
+const GOODS_RECEIVED_2: (&str, &str) = (
+    "3486239A597646B2B7259D91A24988E9",
+    r#"{
+        "ID": "3486239A597646B2B7259D91A24988E9",
+        "budget_ID": "",
+        "comment": "",
+        "donor_id": "",
+        "entry_date": "2025-07-24",
+        "linked_transaction_ID": "",
+        "purchase_order_ID": "",
+        "received_date": "2025-07-24",
```

Comment:
I think we did the deserialisers wrong - and doesn't handle empty string (null) optional dates.

Would you be willing to change received_date deserialiser to this?

I found this on my manual testing - create a goods_received, sync to OG, fails to sync back with empty date.

```
    #[serde(deserialize_with = "zero_date_as_option")]
    #[serde(serialize_with = "date_option_to_isostring")]
    pub received_date: Option<NaiveDate>,
```